### PR TITLE
chore: add GH Actions CI

### PR DESCRIPTION
**Pull request by SimenB**
Sets up GH Actions. Can replace both travis and appveyor.

It's (way) faster and has better integration with GH itself, which might matter if we ever wanna do releases from CI or some such. It also automatically picks up typescript and eslint errors and adds inline annotations, which is nice 🙂

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
 		]
 	},
 	"prettier": {
+		"endOfLine": "auto",
```
Comment (SimenB) [inline package.json]: so it passes on windows. By default prettier reads from `.editorconfig`. We could add a `.gitattributes` file as well

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Node.js CI
+
+on: [push, pull_request]
```
Comment (SimenB) [inline .github/workflows/nodejs.yml]: could do on push to master, and all PRs, so we don't get what happened here - both for PR and for push

File: package.json (hunk 1 above)
Comment (SimenB) [reply package.json]: alternatively, we could have separate jobs for tests (all OSes and node versions) and lint/typescript (LTS on ubuntu)

Comment (zbjornson) [reply package.json]: I'm a bit confused. Is prettier using CRLF on Windows, despite `.editorconfig` currently containing `end_of_line = lf`? Or where are the CRLFs coming from? (My pref is to use LF on all platforms.)

Comment (SimenB) [reply package.json]: GitHub by default checks out with crlf. We can add a gitattributes file as well

Comment (SimenB) [reply package.json]: See https://github.community/t5/GitHub-Actions/git-config-core-autocrlf-should-default-to-false/td-p/30445

Comment (zbjornson) [reply package.json]: I'm +1 on using `.gitattributes`. autocrlf is evil :)

**Comment by sam-github** [comment]
Haven't played with gitactions enough to comment on the config, but moving towards one CI system rather than 2 sounds like a win.

Maybe you want to add the build status badge, too?

https://help.github.com/en/actions/automating-your-workflow-with-github-actions/configuring-a-workflow#adding-a-workflow-status-badge-to-your-repository

**Comment by SimenB** [comment]
Speed wise: 
image

**Comment by siimon** [comment]
Great work and a good suggestion 👍

**Pull request merged**